sympy/categories/baseclasses.py

```diff
@@ -273,11 +273,7 @@ def __new__(cls, *components):
 
         normalised_components = Tuple()
 
-        # TODO: Fix the unpythonicity.
-        for i in range(len(components) - 1):
-            current = components[i]
-            following = components[i + 1]
-
+        for current, following in zip(components, components[1:]):
```

Comment:
Should there be `components[:-1]`?

Reply:
Pythonic as well as a little more fast. I am okay with it.